packages/viz/src/lib/sources/CARTOSource.ts

```diff
@@ -95,10 +96,108 @@ export class CARTOSource extends Source {
     return this._credentials;
   }
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore
-  // eslint-disable-next-line
-  public async getMetadataForField(field: string): Promise<Metadata> {
-    throw new Error('Method not implemented.');
+  public getFieldStats(): NumericFieldStats | CategoryFieldStats {
+    // initialize the stats to 0
+
+    if (!this.isInitialize)
+      throw new Error('To run this method you need to first call to init()');
+
+    if (this._fieldStats === undefined) {
+      throw new Error('Stats are not set after map instantiation');
+    }
+
+    return this._fieldStats;
+  }
+
+  private _initConfigForStats(field: string) {
+    if (this._mapConfig.metadata === undefined) {
+      throw new Error('Map Config has not metadata fiedl');
+    }
+
+    // Modify mapConfig to add the field stats
+    this._mapConfig.metadata.columnStats = {
+      topCategories: 32768,
+      includeNulls: true
+    };
+    this._mapConfig.metadata.dimensions = true;
+
+    /* eslint-disable @typescript-eslint/camelcase */
+    this._mapConfig.metadata.sample = {
+      num_rows: 1000,
+      include_columns: [field]
+    };
+    /* eslint-enable @typescript-eslint/camelcase */
+
+    this._mapConfig.aggregation = {
+      columns: {},
+      dimensions: {
+        [field]: { column: field }
+      },
+      placement: 'centroid',
```

Comment:
Also this 'placement' option had a lot of discussion in the pas. In VL it was fixed, but then we're loosing quite a lot of options supported by our backend. +1 to made it a param in the future